src/plugins/log-errors.js

```diff
@@ -0,0 +1,37 @@
+import Boom from '@hapi/boom'
+
+import { ApplicationError } from '~/src/api/forms/errors.js'
+
+/**
+ * @satisfies {import('@hapi/hapi').Plugin<void>}
+ */
+export const logErrors = {
+  name: 'log-errors',
+  register: (server) => {
+    server.ext('onPreResponse', (request, h) => {
+      const response = request.response
+
+      if (Boom.isBoom(response)) {
+        // An error was raised while processing the request
+        const statusCode = response.output.statusCode
+        const data = response.data ? response.data.stack : response.stack
+
+        // Log the error
+        request.log('error', {
+          statusCode,
+          message: response.message,
+          stack: data
+        })
+
+        if (response instanceof ApplicationError) {
+          response.output.statusCode = response.statusCode
+          response.output.payload.statusCode = response.statusCode
+          response.output.payload.message = response.message
+          response.output.payload.error = response.name
+        }
```

Comment:
This section here adjusts the output from the default 'Internal Server Error' message to the properties from the ApplicationErrors. So a more appropriate name for this file would be something like "transform-error-output" rather than just "log-errors". It does both, but I'd argue this is the most important bit - don't want anyone thinking this is an optional logger that they can turn off, without realising it breaks the API contract.

Reply:
Refactored into separate plugin